### DISCUSSION TITLE
Dynamically enabling required arguments for a command based on the user's input from first argument

### DISCUSSION THREAD

**Discussion by yashika51** [fastapi/typer#1321] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer()


@app.command()
def connection(
    name: str = typer.Option(
        ..., help=typer.style("Name", fg=typer.colors.BRIGHT_GREEN, bold=True)
    ),
    display_name: str = typer.Option(
        ..., help=typer.style("Display Name", fg=typer.colors.BRIGHT_GREEN, bold=True)
    ),
    service_account_json: str = typer.Option(
        ...,
        help=typer.style(
            "JSON enclosed in single quotes", fg=typer.colors.BRIGHT_GREEN, bold=True
        ),
    ),
    ignored_users: str = typer.Option(
        "[]",
        help=typer.style(
            "Ignored users as single quotes enclosed list of strings",
            fg=typer.colors.YELLOW,
        ),
    )
) -> None:
    """Connect the platform"""

    pass
```

### Description

Hello, I am very new to Typer and have a question regarding the possibility of dynamic arguments or any suggestions :) 

In the above code sample, I am trying to connect a platform with the CLI. The platform in example has to have `name`, `display name`, `service account json `and some other optional fields. However, if the command has too many arguments it looks messy and is a lot of work for user in case of positional arguments. 

What I want is to give user options. If a user has a `json` or `yaml` file with config details I want to just read the values from there and disable input for required arguments, otherwise if the user doesn't have the file, they should pass all the arguments using the right flags(-- name xyz for example). 

This can depend on the first argument lets say    
` is_config_file_available: bool = typer.Option(False, help = "Do you want to load credentials from  the config file?"),` 

If the user input here is True then they can pass the file path along and doesn't have to pass any other argument. 

Is this possible to do with Typer? 

### Operating System

macOS

### Typer Version

0.4.0

### Python Version

Python 3.8.10

**Comment by jonaslb** [comment]
Not sure that it's possible with typer directly, but if you have lots and lots of parameters, I can understand the want for a config file. In that case, if you still want a simple type annotated developer experience (which I can sympathize with), you could just use a `Path` argument in typer, and then use something like [Mashumaro](https://github.com/Fatal1ty/mashumaro) to load/deserialize the config file.

**Comment by YuriiMotov** [comment]
I think this can be solve by introducing mutually exclusive options: https://github.com/fastapi/typer/issues/140

If you don't have a lot of commands, you can solve this by adding separate commands that would read parameters from config file and call existing command functions:

```py
@app.command()
def connection(
    name: str = typer.Option(),
    display_name: str = typer.Option(),
    service_account_json: str = typer.Option(),
    ignored_users: str = typer.Option()
) -> None:
    """Connect the platform"""
    pass

@app.command()
def connection_from_config():
    """Connect the platform. Read configuration from file"""
    # read config from file
    connection(name=name_from_config, ...)
```
